files/architecture/security.tex

```diff
@@ -0,0 +1,27 @@
+\chapter{Security}
+This chapter describes the considerations and decisions taken to secure the back-end application.
+
+\section{Authorization}
+To make sure that malicious users can't abuse the back-end application, the decision to secure it through authorization was made. There was 2 options for how this could be handled. 
+
+\subsection{The common approach}
+This approach is to add a token of sorts to the header of the http request. The downside of this option when working with GraphQL is that you can not add a http header to your request in graphiql when you query from the browser, unless a browser extension for that specific thing is used. It would also not be very transparent or obvious when browsing graphiql that a authentication token is required for certain endpoints.
+
+\subsection{The facebook approach}
+Facebook uses a root type called a viewer(). The viewer represent the user making the requests. This can be used for several things. What we did was add a parameter on the root type viewer so that the viewer takes a token. This approach makes it very obvious that the viewer 'area' is restricted and requires a authorization token to access. Every endpoint that requires an authorized user will then be underneath the viewer type and every other endpoint are exposed outside the viewer object. On figure \ref{fig:queryViewer} this is illustrated.
+
+\graphic{0.6}{query}{Query documentation showing the viewer and login}{fig:queryViewer}
+
+Notice that both of them returns a viewer type. When logging in you get the same viewer as you can get when asking for the viewer with your token. This is to save a round-trip after log in. Another approach is to login and only receive a token which you can then use to query for the viewer. But that approach is a lot slower since it will require 2 round-trips.
```

Comment:
This is written very blogish. Try removing to "you"s to a more academic language.